frontend/public/components/masthead.tsx

```diff
@@ -9,12 +10,96 @@ import * as dedicatedLogoImg from '../imgs/openshift-dedicated-logo.svg';
+import { SoftwareDetails } from './software-details';
 
 import { coFetchJSON } from '../co-fetch';
 import { SafetyFirst } from './safety-first';
 
 const developerConsoleURL = (window as any).SERVER_FLAGS.developerConsoleURL;
 
+export class HelpMenu extends React.Component<HelpMenuProps, HelpMenuState> {
+  constructor (props) {
+    super(props);
+    this.state = { showModal: false };
+    this.close = this.close.bind(this);
+    this.open = this.open.bind(this);
+    this.redirect = this.redirect.bind(this);
+  }
+
+  close() {
+    this.setState({ showModal: false });
+  }
+
+  open() {
+    this.setState({ showModal: true });
+  }
+
+  redirect() {
+    window.open('https://docs.openshift.com/online/welcome/index.html', '_blank');
+  }
+
+  render() {
+    const {username} = this.props;
+    return <React.Fragment>
+      <ActionsMenu
+        actions={[
+          {label: 'Documentation', callback: this.redirect},
+          {label: 'About', callback: this.open}]}
+        noButton
+        noCaret
+        title={<i className="fa fa-question-circle-o co-masthead__help-icon" />} />
+      <AboutModal className="co-masthead__modal" logo={okdLogoImg} altLogo="OKD" productTitle="OpenShift" show={this.state.showModal} onHide={this.close}>
+        <h2>About</h2>
+        <p>OpenShift is Red Hat&apos;s container application platform that allows developers to quickly develop, host, and scale applications in a cloud environment.</p>
+        <h2>Version</h2>
+        <SoftwareDetails className="co-masthead__modal--details" kubernetesTitle="Kubernetes Master" noColumns openShiftFirst openShiftTitle="OpenShift Master" titleClassName="co-masthead__modal--subtitle" detailClassName="co-masthead__modal--version" />
+        {username && <React.Fragment>
+          <h2>Account</h2>
+          <p>You are currently logged in under the user account <span className="co-masthead__modal--username">{username}</span>.</p>
+        </React.Fragment>}
+      </AboutModal>
+    </React.Fragment>;
+  }
+}
+
+export class OSHelpMenu extends SafetyFirst<{}, OSHelpMenuState> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: undefined
+    };
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    this._getUserInfo();
+  }
+
+  _getUserInfo() {
+    coFetchJSON('api/kubernetes/apis/user.openshift.io/v1/users/~')
+      .then((user) => {
+        this.setState({ username: _.get(user, 'fullName') || user.metadata.name });
+      }).catch(() => this.setState({ username: null }));
+  }
+
+  render () {
+    const username = this.state.username;
+    return username ? <HelpMenu username={username} /> : null;
+  }
+}
+
+const HelpMenuWrapper = connectToFlags(FLAGS.AUTH_ENABLED, FLAGS.OPENSHIFT)((props: FlagsProps) => {
+  if (flagPending(props.flags[FLAGS.OPENSHIFT]) || flagPending(props.flags[FLAGS.AUTH_ENABLED])) {
+    return null;
+  }
+
+  if (props.flags[FLAGS.OPENSHIFT]) {
+    return <OSHelpMenu />;
```

Comment:
We shouldn't need separate help menus here. The items should be the same regardless of whether it's OpenShift (at least for now).